xarray/core/weighted.py

```diff
@@ -239,6 +291,113 @@ def _weighted_std(
 
         return cast("DataArray", np.sqrt(self._weighted_var(da, dim, skipna)))
 
+    def _weighted_quantile(
+        self,
+        da: "DataArray",
+        q,
+        dim: Optional[Union[Hashable, Iterable[Hashable]]] = None,
+        skipna: Optional[bool] = None,
+        keep_attrs: Optional[bool] = None,
+    ) -> "DataArray":
+        """Apply a weighted ``quantile`` to a DataArray along some dimension(s)."""
+
+        def _get_h(n, q, htype=7):
+            if htype == 4:
+                h = n * q
+            if htype == 5:
+                h = n * q + 0.5
+            if htype == 6:
+                h = (n + 1) * q
+            if htype == 7:
+                h = (n - 1) * q + 1
+            if htype == 8:
+                h = (n + 1 / 3) * q + 1 / 3
+            if htype == 9:
+                h = (n + 1 / 4) * q + 3 / 8
+            return h.clip(1, n)
+
+        def _weighted_quantile_1d(data, weights, q, skipna, htype=7):
```

Comment:
```suggestion
        def _weighted_quantile_1d(data, weights, q, skipna: bool, htype: Literal[7, 4, 5, 6, 8, 9] =7):
```
Might have to import Literal from typing as well.